CONTRIBUTING.md

```diff
@@ -0,0 +1,113 @@
+
+Once you have selected an issue you'd like to work on, make sure it's not already assigned to someone else, and assign it to yourself.
+
+It is recommended that you use a separate branch for every issue you work on. To keep things straightforward and memorable, you can name each branch using the GitHub issue number. This way, you can have multiple PRs open for different issues. For example, if you were working on [issue-125](https://github.com/wise-agents/wise-agents/issues/125), you could use issue-125 as your branch name.
+
+## Setting up your Developer Environment
+You will need:
+
+* Python 3.12+
+* Git
+* An [IDE](https://en.wikipedia.org/wiki/Comparison_of_integrated_development_environments#Python)
+(e.g., [Microsoft Visual Studio Code](https://code.visualstudio.com/))
+
+To setup your development environment you need to:
+
+1. First `cd` to the directory where you cloned the project (eg: `cd wise-agents`)
+
+2. Create a Python wirtual environment for the project.
+
+    The venv module supports creating lightweight “virtual  environments”, each with their own independent set of Python packages installed in their site directories. A virtual environment is created on top of an existing Python installation, known as the virtual environment’s “base” Python, and may optionally be isolated from the packages in the base environment, so only those explicitly installed in the virtual environment are available.
+For more information about virtual environment see [here](https://docs.python.org/3/library/venv.html)
+
+    ```
+    python -m venv ./venv
+    ```
+
+3. Activate the venv
+    ```
+    source .venv/bin/activate
+    ```
+4. Add a remote ref to upstream, for pulling future updates.
+For example:
+
+    ```
+    git remote add upstream https://github.com/wise-agents/wise-agents
+    ```
+5. To build `wise-agents` run:
+    ```bash
+    TODO: waiting for makefile
+    ```
+
+6. To skip the tests, use:
+
+    ```bash
+    TODO: waiting for makefile
+    ```
+
+7. To run only a specific test, use:
+
+    ```bash
+    TODO: waiting for makefile
+    ```
+
+
+## Contributing Guidelines
+
+When submitting a PR, please keep the following guidelines in mind:
+
+1. In general, it's good practice to squash all of your commits into a single commit. For larger changes, it's ok to have multiple meaningful commits. If you need help with squashing your commits, feel free to ask us how to do this on your pull request. We're more than happy to help!
+
+2. Please include the issue you worked on in the title of your pull request and in your commit message. For example, for [issue-125](https://github.com/wise-agents/wise-agents/issues/125), the PR title and commit message should be `Go through TODOs in the code and create issues for them #125`.
```

Comment:
For GitHub issues, unlike the standard practice with JIRAs, it seems that most projects don't ask for the issue number in the commit message/title.

Instead, a pull request can be linked to an issue using a keyword (as described here: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

So maybe we can reword 2. to something like:

Please link the issue you worked on in the description of your pull request and in your commit message. For example, for [issue-125](https://github.com/wise-agents/wise-agents/issues/125), the PR description and commit message could be: 

```
Go through TODOs in the code and create issues for them

Fixes #125
```